### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by nicolechung**
screen shot 2017-10-17 at 6 38 40 am

Added a test for the Error Boundary. Not sure if it's legit. I tried to render a `throw new Error('broken')` but jest did not seem to like that.

Updated to react 16, also react-dom, and react-test-renderer.

## To test
[ ] - Please clear node_modules and do a `yarn install`
[ ] - `yarn test`

### CODE REVIEW

**Comment by nicolechung** [comment]
@hyperwidget Should we add something like `yarn test` to the `before deploy` in `travis.yml`?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "standard.enable": false,
+  "editor.formatOnSave": false
+}
```
Comment (spiderbites) [inline .vscode/settings.json]: I don't think this file should be in version control

File: jest/testSetup.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+})()
+
+Object.defineProperty(window, 'localStorage', {
+  value: localStorageMock
+})
+
+Object.defineProperty(window.location, 'href', {
+  writable: true,
+  value: 'some url'
+})
```
Comment (spiderbites) [inline jest/testSetup.js]: I don't think we need this in here, it's used specifically for testing window.location.href, but i don't think that applies to tests in this repo

**Comment by spiderbites** [comment]
good stuff!!  💯

File: .vscode/settings.json (hunk 1 above)
Comment (nicolechung) [reply .vscode/settings.json]: Okay let me remove it..

Comment (hyperwidget) [reply .vscode/settings.json]: I'd just add it to gitignore

**Comment by nicolechung** [comment]
Github is not showing my new commits. I will check in an hour and let you know if these changes are up.

**Comment by nicolechung** [comment]
@spiderbites changes are up

**Comment by spiderbites** [comment]
@nicolechung i'm just trying to figure out why travis is not running against this project, looks like it stopped a couple PRs ago

**Pull request merged**